new(mc): added new telegraf connector (#3354)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/operatingsystems-windows-telegraf-agent.md

| sessions.reconnected.total.count    | count |
| sessions.active.current.count       | count |
| sessions.disconnected.current.count | count |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
<TabItem value="Updates" label="Updates">

| Metric name                   | Unit  |
|:------------------------------|:------|
| windows.pending.updates.count | count |

</TabItem>
</Tabs>

## Prerequisites

### Centreon Engine

> To be able to use the Telegraf agent, you must use a poller with at least version `24.04.2` of `centreon-engine`.

The Telegraf agent will configure itself via a HTTPS request sent to Centreon Engine.

1. For this to work, you must first generate a certificate on the poller:

```bash
openssl req -new -newkey rsa:2048 -sha256 -days 3650 -nodes -x509 -keyout /etc/centreon-engine/conf-server.key -out /etc/centreon-engine/conf-server.crt
chown centreon-engine: /etc/centreon-engine/conf-*
```

> **Important** : give the IP address or the FQDN of the poller when the **Common Name** information is prompted.
>
> Example:
> ```
> Common Name (e.g. server FQDN or YOUR name) []:172.16.5.166
> ```

> 2. Then the configuration server must be set up
> 
> Prepare the information Engine will have to provide to the Telegraf agent.

Author: ```suggestion
2. Then prepare the information Engine will have to provide to the Telegraf agent.
```

```bash
openssl req -new -newkey rsa:2048 -sha256 -days 3650 -nodes -x509 -keyout /etc/centreon-engine/conf-server.key -out /etc/centreon-engine/conf-server.crt
chown centreon-engine: /etc/centreon-engine/conf-*
```

> **Important** : give the ip address or the FQDN of the poller when the **Common Name** information is prompted.
>
> Example:
> ```
> Common Name (e.g. server FQDN or YOUR name) []:172.16.5.166
> ```

2. Then the configuration server must be set up

Prepare the information Engine will have to provide to the Telegraf agents.

```bash
host_ip=$(hostname -I)
cat > /etc/centreon-engine/otl_server.json <<EOF
{
    "otel_server": {
        "host": "0.0.0.0",
        "port": 4317
    },
    "max_length_grpc_log": 0,
    "telegraf_conf_server": {
        "http_server" : {
            "port": 1443,
            "encryption": true,
            "certificate_path": "/etc/centreon-engine/conf-server.crt",
            "key_path": "/etc/centreon-engine/conf-server.key"
        },
        "engine_otel_endpoint": "${host_ip% }:4317",
        "check_interval":60
    }
}
EOF
chown centreon-engine: /etc/centreon-engine/otl_server.json
systemctl restart centengine
```

### Telegraf agent

> The official installation procedure is [here](https://docs.influxdata.com/telegraf/v1/install/?t=Windows) but you'll find the main steps here.

Reviewer: ```suggestion
The official installation procedure is [here](https://docs.influxdata.com/telegraf/v1/install/?t=Windows) but you'll find the main steps below.
```

#### Installing the Telegraf agent

> 1. Downloading and unarchiving

Reviewer: ```suggestion
1. Download and unarchive the agent.
```

```powershell
wget https://dl.influxdata.com/telegraf/releases/telegraf-1.30.1_windows_amd64.zip -UseBasicParsing -OutFile telegraf-1.30.1_windows_amd64.zip
Expand-Archive .\telegraf-1.30.1_windows_amd64.zip -DestinationPath 'C:\Program Files\InfluxData\telegraf\'
```

> 2. Installation of the service

Reviewer: ```suggestion
2. Install the service:
```

```cmd
"C:\Program Files\InfluxData\telegraf\telegraf-1.30.3\telegraf.exe" --config https://<poller_ip_address>:1443/engine?host=<windows_server_name> --service install
```

#### Downloading centreon_plugins.exe

Download the **centreon_plugins.exe** from the latest release available [here](https://github.com/centreon/centreon-nsclient-build/releases)

> and copy it at the same place as the Telegraf agent (that should be `"C:\Program Files\InfluxData\telegraf\telegraf-1.30.3\"`)

Reviewer: ```suggestion
and copy it in the same place as the Telegraf agent (that should be `"C:\Program Files\InfluxData\telegraf\telegraf-1.30.3\"`)
```

if you followed the procedure to the letter. 

#### Import du certificat du poller

> 1. Open Edge web browser.

Reviewer: ```suggestion
1. Open the Edge web browser.
```

> 2. Browse https://<poller_ip_address>:1443

Reviewer: ```suggestion
2. Browse to `https://<poller_ip_address>:1443`.
```

> 3. Accept the certificate.
> 4. `No host service found from get parameters` should be displayed.

Reviewer: ```suggestion
3. Accept the certificate. `No host service found from get parameters` should be displayed.
```

5. Click on the certificate on the left of the URL bar.
6. Go to **Details**.
7. Export the certificate.
8. Open the file explorer.

> 9. Right click on the certificate at the place where you downloaded it.

Reviewer: ```suggestion
9. Right-click on the certificate where you downloaded it.
```

> 10. Select Install the certificate.

Reviewer: ```suggestion
10. Select **Install the certificate**.
```

11. Select **Local machine** then **Next**.
12. Specify you want to choose the certificate store manually.
13. Select **Trusted Root Certificate Authority** then **OK**.

> 14. Click on **Next** puis sur **Finish**.

Reviewer: ```suggestion
14. Click **Next**, then  **Finish**.
```

instruction below as it is not required to have the connector displayed within the
**Configuration > Monitoring Connectors Manager** menu.
If the platform uses an *offline* license, install the package on the **central server**
with the command corresponding to the operating system's package manager:

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-pack-operatingsystems-windows-telegraf-agent
```

</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf install centreon-pack-operatingsystems-windows-telegraf-agent
```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

```bash
apt install centreon-pack-operatingsystems-windows-telegraf-agent
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-pack-operatingsystems-windows-telegraf-agent
```

</TabItem>
</Tabs>

2. Whatever the license type (*online* or *offline*), install the **Windows Telegraf Agent** connector through
the **Configuration > Monitoring Connectors Manager** menu.

> 3. Connector to associate with commands

Reviewer: ```suggestion
3. Create the corresponding connector:
```

> In the **Configuration > Commandes > Connecteurs** menu, click **Add** and fill the following fields:

Reviewer: ```suggestion
In the **Configuration > Commands > Connectors** menu, click **Add** and fill the following fields:
```

| Connector Name         | Telegraf Agent                                                                                                                                                                                                                   |
| Connector Description | Telegraf Agent                                                                                                                                                                                                                   |
| Command Line         | `opentelemetry --processor=nagios_telegraf --extractor=attributes --host_path=resourceMetrics.scopeMetrics.metrics.dataPoints.attributes.host --service_path=resourceMetrics.scopeMetrics.metrics.dataPoints.attributes.service` |
| Used by command   | Select all the commands named like `OS-Windows-Telegraf-Agent-*`                                                                                                                                                                  |

> | Connector Status     | Activé                                                                                                                                                                                                                           |

Reviewer: ```suggestion
| Connector Status     | Enabled                                                                                                                                                                                                                          |
```
